src/ProxyClient/Varnish.php

```diff
@@ -60,14 +60,17 @@ class Varnish extends AbstractProxyClient implements BanInterface, PurgeInterfac
 
     /**
      * {@inheritdoc}
+     *
+     * @param string $tagsHeader Header for tags, defaults to X-Cache-Tags
      */
     public function __construct(
         array $servers,
         $baseUri = null,
-        HttpAdapter $httpAdapter = null,
+        HttpAsyncClient $httpClient = null,
+        MessageFactory $messageFactory = null,
         $tagsHeader = 'X-Cache-Tags'
```

Comment:
we could, but baseUri is something of the base client. do we want to mix
base client options and specific options? though maybe we do want, as
its optional and one day we might find more optional things for the base
client, at which point it would become tricky with parameter order. i
would keep http client, message factory and servers out. servers because
they are required, the client and factory because they seem more than a
simple option. agreed?